### PR TITLE
Shub pull panic fix

### PR DESCRIPTION
**Pull request by ikaneshiro**
Fixes the panic caused by a nil http response, output below:

```DEBUG[U=2000,P=15176]   Get()                         Getting container from Shub
DEBUG[U=2000,P=15176]   NewBundle()                   Created temporary directory for bundle /tmp/sbuild-shub-103409937
DEBUG[U=2000,P=15176]   DownloadImage()               Downloading container from Shub
--- FAIL: TestShubConveyor (30.01s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xc4dd6a]
goroutine 174 [running]:
testing.tRunner.func1(0xc4202ea0f0)
	/home/travis/.gimme/versions/go1.10.4.linux.amd64/src/testing/testing.go:742 +0x567
panic(0xd646c0, 0x1488240)
	/home/travis/.gimme/versions/go1.10.4.linux.amd64/src/runtime/panic.go:502 +0x24a
github.com/singularityware/singularity/src/pkg/shub/client.getManifest(0xe48f3e, 0x2a, 0xc42040a007, 0xa, 0xc42040a012, 0xe, 0xc4203b02a0, 0x7, 0x0, 0x0, ...)
	/home/travis/gopath/src/github.com/singularityware/singularity/src/pkg/shub/client/api.go:81 +0x86a
github.com/singularityware/singularity/src/pkg/shub/client.DownloadImage(0xc42040a030, 0x2c, 0xc42040a000, 0x27, 0xc4200b6301, 0x0, 0x0)
	/home/travis/gopath/src/github.com/singularityware/singularity/src/pkg/shub/client/pull.go:51 +0x2ba
github.com/singularityware/singularity/src/pkg/build/sources.(*ShubConveyorPacker).Get(0xc420791e80, 0xc420136000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/travis/gopath/src/github.com/singularityware/singularity/src/pkg/build/sources/conveyorPacker_shub.go:48 +0x4a0
github.com/singularityware/singularity/src/pkg/build/sources_test.TestShubConveyor(0xc4202ea0f0)
	/home/travis/gopath/src/github.com/singularityware/singularity/src/pkg/build/sources/conveyorPacker_shub_test.go:37 +0x2c1
testing.tRunner(0xc4202ea0f0, 0xe5f9a0)
	/home/travis/.gimme/versions/go1.10.4.linux.amd64/src/testing/testing.go:777 +0x16e
created by testing.(*T).Run
	/home/travis/.gimme/versions/go1.10.4.linux.amd64/src/testing/testing.go:824 +0x565```


Also, some comment formatting fixes

### CODE REVIEW

File: src/pkg/shub/client/api.go [hunk 1]
```diff
@@ -78,6 +78,9 @@ func getManifest(uri ShubURI) (manifest ShubAPIResponse, err error) {
 
 	// Do the request, if status isn't success, return error
 	res, err := httpc.Do(req)
+	if res == nil {
+		return ShubAPIResponse{}, fmt.Errorf("No response received from singularity hub")
+	}
```
Comment (ArangoGutierrez) [inline src/pkg/shub/client/api.go:83]:
```go
	if resp.StatusCode == http.StatusNotFound {
		return fmt.Errorf("The requested image was not found in singularity hub")
	}
```
as suggested by @tri-adam

File: src/pkg/shub/client/pull.go [hunk 2]
```diff
@@ -66,6 +66,9 @@ func DownloadImage(filePath string, shubRef string, force bool) (err error) {
 
 	// Do the request, if status isn't success, return error
 	resp, err := httpc.Do(req)
+	if resp == nil {
```
Comment (ArangoGutierrez) [inline src/pkg/shub/client/pull.go:69]: same as above

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]

File: src/pkg/shub/client/api.go (hunk 1 above)
Comment (ikaneshiro) [reply src/pkg/shub/client/api.go:83]: That check is already there, I think he was saying that not having this direct check of `res == nil` and just going straight into a `res.StatusCode` is what's causing the panic in the case where `res` is `nil`. @tri-adam correct me if im wrong though

Comment (ArangoGutierrez) [reply src/pkg/shub/client/api.go:83]: sorry, didn't read it all. my bad

**Review by ArangoGutierrez — APPROVED** [review]

File: src/pkg/shub/client/api.go (hunk 1 above)
Comment (ikaneshiro) [reply src/pkg/shub/client/api.go:83]: no worries

**Comment by ArangoGutierrez** [comment]
@bauerm97  RFR

**Review by jmstover — APPROVED** [review]

File: src/pkg/shub/client/api.go [hunk 3]
```diff
@@ -78,6 +78,9 @@ func getManifest(uri ShubURI) (manifest ShubAPIResponse, err error) {
 
 	// Do the request, if status isn't success, return error
 	res, err := httpc.Do(req)
+	if res == nil {
+		return ShubAPIResponse{}, fmt.Errorf("No response received from singularity hub")
```
Comment (GodloveD) [inline src/pkg/shub/client/api.go:82]: Why using `fmt` instead of `sylog`?

**Review by GodloveD — DISMISSED** [review]

File: src/pkg/shub/client/api.go (hunk 3 above)
Comment (ArangoGutierrez) [reply src/pkg/shub/client/api.go:82]: he is returning an error type, to be handled else were, on the error handling step, it will go into a sylog.

**Pull request merged**